### ISSUE TITLE
SagemakerNotebookInstance not deleted

### ISSUE THREAD

**Issue by jcataluna** [rebuy-de/aws-nuke#192]
Hi,

with a targets configuration like:

  - SageMakerNotebookInstanceState
  - SageMakerNotebookInstance
  - SageMakerModel
  - SageMakerEndpoint
  - SageMakerEndpointConfig

notebooks are stopped but not deleted. I've tried to remove SageMakerNotebookInstanceState from targets but instance is not deleted because it's not in stopped state.

Regards

**Comment by bjoernhaeuser** [comment]
Hi @jcataluna 

Can we see your configuration (please remove private information from it!) and also the log output of the nuke run?

**Comment by jcataluna** [comment]
Sure ;)

```
regions:
- us-east-2
- us-east-1	
- us-west-1	
- us-west-2	
- ap-south-1
- ap-northeast-2
- ap-northeast-3
- ap-southeast-1
- ap-southeast-2
- ap-northeast-1
- ca-central-1
- cn-north-1
- eu-central-1
- eu-west-1
- eu-west-2
- eu-west-3	
- sa-east-1

account-blacklist:
- "999999999999" # production


resource-types:
  # don't nuke IAM users
  targets:
  - CloudWatchLogsLogGroup
  - APIGatewayRestAPI
  - SageMakerNotebookInstanceState
  - SageMakerNotebookInstance
  - SageMakerModel
  - SageMakerEndpoint
  - SageMakerEndpointConfig
  - EC2Instance
  - EC2Volume
  - S3Bucket
  - S3Object
  - LambdaFunction
  - WorkSpacesWorkspace
  - RedshiftCluster
  - RedshiftParameterGroup
  - RedshiftSnapshot
  - RedshiftSubnetGroup
```
Regarding logs it's not possible because accounts are clean right now. But the "SageMakerNotebookInstance" step got stuck on waiting state while the instance it's not stopped. If at that time the instance it's deleted from console, obviously the lock disappear.

To solve the problem and previously to run aws-nuke,  I stopped the instances with the following script:

```
for reg in $regions; do
    echo "Regions: $reg"
    for i in $(seq -f "%02g" 6 35); do
        echo "acc: customer$i"
        instances=$(aws --profile customer$i --region $reg sagemaker list-notebook-instances --query 'NotebookInstances[*].NotebookInstanceName' | tr -d '"[],' | sed '/^\s*$/d')
        for inst in $instances; do
            echo "Instance: $inst"
            aws --profile customer$i --region $reg sagemaker stop-notebook-instance --notebook-instance-name $inst
        done
    done
done

```

**Comment by svenwltr** [comment]
Hello @jcataluna. Thank you for your issue. This problem will be fixed in #193 and I'll create a new release as soon as it gets merged.
